clio-server/src/main/scala/org/broadinstitute/clio/server/service/IndexService.scala

```diff
@@ -45,26 +45,51 @@ abstract class IndexService[CI <: ClioIndex](
     metadata: clioIndex.MetadataType,
     force: Boolean = false
   ): Source[UpsertId, NotUsed] = {
-    // query by key to see if the document already exists
-    if (!force) {
-      queryMetadataForKey(indexKey)
-        .fold[Either[Exception, clioIndex.MetadataType]](
-          // If the query doesn't return a document and the document status is null then set it to the default value.
-          Right(
-            setDefaultDocumentStatus(metadata)
-          )
-        ) { (_, existingMetadataJson) =>
-          // validation will check to make sure no fields are being overwritten
-          validateUpsert(existingMetadataJson, metadata)
-        }
-        .flatMapConcat[UpsertId, NotUsed] {
-          case Right(updatedMetadata) =>
-            upsertToStream(indexKey, updatedMetadata)
-          case Left(rejection) => Source.failed(rejection)
+    // query by key to see if the document already exists if force is false
+    val validateOrError =
+      if (!force) {
+        queryMetadataForKey(indexKey)
+          .fold[Either[Exception, clioIndex.MetadataType]](
+            // If the query doesn't return a document and the document status is null then set it to the default value.
+            Right(
+              setDefaultDocumentStatus(metadata)
+            )
+          ) { (_, existingMetadataJson) =>
+            //if the incoming document status is not set then set it to the existing status
+            existingMetadataJson
+              .as[clioIndex.MetadataType]
+              .flatMap { existing =>
+                validateUpsert(
+                  existing,
+                  metadata.withDocumentStatus(existing.documentStatus)
```

Comment:
Hrm... will this cause silent failures to overwrite document status? i.e. if the existing doc has status `Normal` and we upsert status `Deleted` with `force=false`, this will overwrite `Deleted` -> `Normal` before validation, so validation will pass, but the status-change will fail to be applied.

Reply:
ugh.. good catch.

Reply:
I must be missing a unit test that would catch this.